server/src/main/java/org/candlepin/dto/StandardTranslator.java

```diff
@@ -188,12 +201,16 @@ public StandardTranslator(ConsumerTypeCurator consumerTypeCurator,
             new ProductCertificateTranslator(), ProductCertificate.class, ProductCertificateDTO.class);
         this.registerTranslator(
             new RoleTranslator(), Role.class, RoleDTO.class);
+        this.registerTranslator(
+            new RoleInfoTranslator(), RoleInfo.class, RoleDTO.class);
```

Comment:
I'm a bit unsure how the *Info classes are going to be translated. Surely when the DTO framework is calling `getClass()` on the input, it will always resolve to `Role` (because that will be the concrete class of the instance at runtime), and not `RoleInfo` (which is just an interface), resulting in always using the `RoleTranslator` and never the 'RoleInfoTranslator'. So the `*InfoTranslator` classes are never going to be actually used. Unless I'm missing something (I might, because I'm not familiar with the adapters at all).

Reply:
You're correct in the one-off translation case where we provide the instance and the destination class, but where this comes into play is when calling the getStreamMappers method, which requires the source class as a parameter.

I suppose this is a use case for adding an overload to .translate which accepts a source class for doing explicit translation. But that can come in another PR that could also address a bug in the lookup for object translators (both input and output scan "up" the hierarchy, but output should scan "down" or we break).